lib/angular2/shared/models/flref.ejs

```diff
@@ -203,10 +257,16 @@ export class FireLoopRef<T> {
       parent: this.parent && this.parent.instance ? this.parent.instance : null
     };
     this.socket.emit(event, config);
-    this.socket.onZone(`${ this.model.getModelName() }.value.result.${ this.id }`, (res: any) =>
-      subject.next(res.error ? Observable.throw(res.error) : res)
-    );
-    return subject.asObservable();
+    this.socket.on(`${ this.model.getModelName() }.value.result.${ this.id }`, (res: any) => {
+      if (res.error) {
+        subject.error(res);
+      } else {
+        subject.next(res);
+      }
+    });
+    // This event listener will be wiped within socket.connections
+    this.socket.sharedObservables.sharedOnDisconnect.subscribe(() => subject.complete());
+    return subject.asObservable().catch((error: any) => Observable.throw(error));
```

Comment:
This is  a rxjs thing. Subejcts are terminated on errors and on completed. 

This is a realtime platform and we dont want to dtop a channel.of communication just because you got an error.

If you simply use the subject.error method, your subscribers wont receive any further message from the server using this reference.

I hope this clarifies your questions